### ISSUE TITLE
Uncaught TypeError: Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by richardsondx** [opentok/opentok-web-samples#80]
My Opentok setup used to work but recently I started getting an error. The browser I'm using is:
-Chrome Version 69.0.3497.100 (Official Build) (64-bit)
macOS Sierra Version 10.12 Beta
ruby '2.5.1
gem 'opentok', '> 3.0.3'
gem 'rails', '> 5.2.1'

Please see below the error I'm getting:

index.js:20 Uncaught TypeError: Cannot convert undefined or null to object
    at slice (<anonymous>)
    at nodeListToArray (index.js:20)
    at selectorToElementArray (index.js:70)
    at new ElementCollection (index.js:103)
    at module.exports (shorthandSelector.js:6)
    at Object.browserAjax.request (browser.js:52)
    at Object.browserAjax.post (browser.js:95)
    at analytics.js:17
    at loop (QueueRunner.js:22)
    at QueueRunner.run (QueueRunner.js:26)

nodeListToArray | @ | index.js:20
-- | -- | --
  | selectorToElementArray | @ | index.js:70
  | ElementCollection | @ | index.js:103
  | module.exports | @ | shorthandSelector.js:6
  | browserAjax.request | @ | browser.js:52
  | browserAjax.post | @ | browser.js:95
  | (anonymous) | @ | analytics.js:17
  | loop | @ | QueueRunner.js:22
  | run | @ | QueueRunner.js:26
  | add | @ | QueueRunner.js:9
  | post | @ | analytics.js:17
  | Analytics.logEvent | @ | analytics.js:130
  | (anonymous) | @ | analytics.js:75
  | (anonymous) | @ | guid_storage.js:118
  | get | @ | guid_storage.js:126
  | getClientGuid | @ | guid_storage.js:30
  | guidStorage.get | @ | guid_storage.js:96
  | logEvent | @ | analytics.js:66
  | Session.logEvent | @ | index.js:531
  | Session.connect | @ | index.js:768
  | (anonymous) | @ | 33:255
when I run this code:

  // Connect to the Session using a 'token'
  session.connect(token, function(error) {
    if (error) {
      console.log("Error connecting: ", error.name, error.message);
    } else {
    debugger
    var publisher = OT.initPublisher('publisher-div', {
                                              frameRate: 80,
                                              insertMode: 'append',
                                              width: '300px',
                                              height: '220px'
                                            }
                                    );
      session.publish(publisher);
      console.log("Connected to the session.");
    }
  });
The error comes from this file: webpack:///./src/js/common-js-helpers/elementCollection/index.js

// A helper for converting a NodeList to a JS Array
const nodeListToArray = function nodeListToArray(nodes) {
  if (env.name !== 'IE' || env.version > 9) {
    return Array.prototype.slice.call(nodes);  // <=---- This line returns the error because nodes is undefined
  }

  // IE 9 and below call use Array.prototype.slice.call against
  // a NodeList, hence the following
  const array = [];

  for (let i = 0, num = nodes.length; i < num; ++i) {
    array.push(nodes[i]);
  }

  return array;
};
My token, api and sessionId are already set up and are correct.

**Comment by VOctavian** [comment]
This works for me 
`if (env.name !== 'IE' || env.version > 9) {
                    if(typeof nodes == 'undefined' || typeof nodes == 'null' ){
                        return [];
                    }
                    return Array.prototype.slice.call(nodes);
                }`

**Comment by richardsondx** [comment]
@VOctavian Thanks! but the problem is coming from Opentok.js and they are expecting us to call the endpoint directly from their server... This need to be fixed in the code.

+1

**Comment by shakalya195** [comment]
Please help. I am having the same issue.

**Comment by aullman** [comment]
Thanks for the heads up about pace.min.js this made me able to reproduce the issue.

It turns out the issue is that pace.js is overriding the global XMLHttpRequest object [here](https://github.com/HubSpot/pace/blob/4afddfb37aec45cc8588a91416fe28ca173314a8/pace.coffee#L343).

We should still be able to work around this issue. It's really bad practice of pace.js though to override globals like that. I have filed a bug for this internally though and we'll work on a fix.

**Comment by aullman** [comment]
I actually did some debugging and figured out how to resolve the issue so that we are creating a new XMLHttpRequest and checking `instanceof (new XMLHttpRequest()).constructor` but after doing that I ran into other issues with the WebSocket because pace.js are also overriding that!

I think that you're going to have to stop using pace.js or ask them to fix the issue and tell them not to override globals. Specifically window.XMLHttpRequest and window.WebSocket.
